### ISSUE TITLE
no-shadow: same-line `eslint-disable-line` placed before the violation is not matched — error fires and directive is reported unused

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25633]
## Summary

A `/* eslint-disable-line no-shadow */` comment on the same physical line as the violation fails to suppress the diagnostic when the comment sits earlier on the line than the reported span (common on wrapped statements, where the directive opens the continuation line). oxlint both emits the `no-shadow` error and reports the directive as "Unused eslint-disable directive". Moving the identical comment to the end of the line (after the violation) suppresses correctly, so the disable-line matcher appears to only associate diagnostics that begin before the comment — ESLint matches by line only, regardless of column. Migrations from ESLint silently lose such suppressions, producing both spurious errors and spurious unused-directive warnings.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.tsx` — the `.map(...)` call and its shadowing `match` parameter share physical line 9 with the directive:

```tsx
export function commandsFromSVG(path: string) {
	const markerRegEx = /[A-Za-z]/g
	let match: RegExpExecArray | null
	const results: RegExpExecArray[] = []
	while ((match = markerRegEx.exec(path)) !== null) {
		results.push(match)
	}
	return results
		/* eslint-disable-line no-shadow */ .map((match: RegExpExecArray) => ({
			marker: path[match.index]!,
			index: match.index,
		}))
}
```

`.oxlintrc.json`:

```json
{
	"plugins": [],
	"categories": {"correctness": "off"},
	"rules": {"no-shadow": ["error"]}
}
```

```
oxlint -c .oxlintrc.json --report-unused-disable-directives repro.tsx
```

## Expected

The directive suppresses the `no-shadow` violation on line 9: no error, no unused-directive warning, exit 0. ESLint 7 and ESLint 9 (core `no-shadow` under the TypeScript parser, `--report-unused-disable-directives`) both honor it. Controls: removing the directive makes ESLint report the shadow at 9:9, and a genuinely unused directive is still flagged — so the flag itself works.

## Actual

```
repro.tsx:9:45: error eslint(no-shadow): 'match' is already declared in the upper scope. help: Consider renaming 'match' to avoid shadowing the variable from the outer scope.
repro.tsx:9:3: warning: Unused eslint-disable directive (no problems were reported).
```

Exit 1. The diagnostic is anchored at the shadowing identifier (9:45), textually after the comment (9:3). Placing the same comment at the end of line 9 instead (`… => ({ /* eslint-disable-line no-shadow */`) suppresses correctly on 1.78.0, isolating the defect to comment-before-diagnostic matching rather than the rule.

## Cross-checked

| Linter | Rule source | Result |
| --- | --- | --- |
| ESLint 7.23.0 (`@typescript-eslint/parser` 6.21.0) | core `no-shadow` | directive honored — no error, no unused report, exit 0 |
| ESLint 9.39.4 flat config (`typescript-eslint` 8.62.1) | core `no-shadow` | directive honored — no error, no unused report, exit 0 |
| oxlint 1.78.0 | built-in `no-shadow` | error at 9:45 + directive at 9:3 reported unused, exit 1 |

**Comment by mark-omarov** [comment]
I would like to give this a shot if I may. I am confident I understand the issue, current implementation, and the potential fix is in the mentioned PR above.
